dynamicProgramming/longestCommonSubsequence.py

```diff
@@ -0,0 +1,71 @@
+"""
+Given 2 sequences, find the length of the longest subsequence present in both of them.
+NOT consecutive. order is important. NOT substring. first brute force it by calculating all permutations that maintains the correct order. Analyze it.
+Then do it with recursion/DP.
+Cache will help. 
+max length of each sequence is 1 billion. Which means you need a good runtime, not EVERY permutation. 
+
+
+Inputs:
+- seq1, string
+- seq2, string
+Output:
+- integer
+"""
+class LCS:
+    def lcs_no_dp(self, seq1, seq2):
+        
+        if len(seq1) < 1:
+            return 0
+
+        if len(seq2) < 1:
+            return 0
+
+        if seq1[0] == seq2[0]:
+            return 1 + self.lcs_no_dp(seq1[1:], seq2[1:])
+
+        else:
+            return max(self.lcs_no_dp(seq1[1:], seq2), self.lcs_no_dp(seq1, seq2[1:]))
+
+    def lcs_memo(self, seq1, seq2, cache):
+        if cache is None:
+            cache = {}
+
+        #Read from the cache
+        #We are going to cache based on length of the strings
```

Comment:
Got the idea to cache based on length of strings from the coin change problem we did before